### ISSUE TITLE
Eliminating Vars in typechecker

### ISSUE THREAD

**Issue by jstolarek** [links-lang/links#733]
Typechecker always converts `Var`s to instantiated frozen variables, except for the cases when there is nothing to instantiate, e.g. `let x =1 in x`. @frank-emrich suggests that we might always convert `Var`s to frozen variables (even if there are no quantifiers to freeze) and then in `sugartoir` assert that no `Var`s should exist in the AST.

**Comment by slindley** [comment]
Yes. That's exactly what we should do. Unfrozen variables are syntactic sugar that should be removed during type inference.
